src/bulk/common.ts

```diff
@@ -725,8 +725,8 @@ export class FindOperators {
     this.bulkOperation = bulkOperation;
   }
 
-  /** Add a multiple update operation to the bulk operation */
-  update(updateDocument: Document): BulkOperationBase {
+  /** @internal */
+  makeUpdateDocument(u: Document, multi: boolean): Document {
```

Comment:
> Can we reuse [this internal helper](https://github.com/mongodb/node-mongodb-native/blob/master/src/operations/update.ts#L250-L254) I introduced last week?

Nice, I refactored to use the `makeUpdateStatement` and `makeDeleteStatement` helpers. Note that there were already helpers by that name in this file, which do a very similar thing but with slightly different behavior.

It would be nice to consolidate them, but for now I've renamed them to `makeBulkUpdateStatement` and `makeBulkDeleteStatement`. There are a few TODOs in there so I want to be careful about replacing them:

```
// NOTE: legacy support for a raw statement, consider removing
// TODO: this check should be done at command construction against a connection, not a topology
```

Regarding removing the legacy support for raw statements, would you have any concerns about that? `4.0` would probably be a good time for it.

Reply:
I went ahead with that consolidation, removed the legacy support for raw statements and fixed the tests we had using the legacy format.

Reply:
Heck'n YES, I love it! I'd say make sure you take another trip through the API docs and tests to ensure that you're properly documenting the shape of what's accepted in bulk. Also, now that the "raw" form of the operations are no longer accepted I think you will want to add some validation to let users know that things have changed. Right now I think you'll be silently breaking their applications potentially, so good to add some tests that prove their apps will not break. Otherwise, I think this is a huge improvement for bulk, great job.